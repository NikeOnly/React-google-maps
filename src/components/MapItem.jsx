import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

class MapItem extends Component {
    static PpropTypes = {
        center: PropTypes.shape({}).isRequired,
        zoom: PropTypes.number.isRequired,
        placesList: PropTypes.arrayOf(PropTypes.object)
    }

    static defaultProps = {
        placesList: []
    }

    render() {
        const { center, zoom, placesList } = this.props;

        return (
            <GoogleMap
                center={center}
                zoom={zoom}
            >
                {placesList && placesList.length ? placesList.map(item => (
                    <Marker
                        key={item.id}
                        position={item.coordinates}
                    />
                )) : null}
            </GoogleMap>
        );
    }
}

export default withScriptjs(withGoogleMap(MapItem));
