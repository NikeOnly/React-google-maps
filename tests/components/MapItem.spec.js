import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import MapItem from '../../src/components/MapItem';

describe('MapItem', () => {
    test('match snapshot', () => {
        const container = shallow(
            <MapItem
                loadingElement={<div />}
                googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
            />
        );

        expect(toJson(container)).toMatchSnapshot();
    })
})
