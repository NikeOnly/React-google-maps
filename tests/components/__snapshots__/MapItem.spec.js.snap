// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapItem match snapshot 1`] = `<div />`;
